src/adapters/CKEditorAdapter.ts

```diff
@@ -20,22 +20,45 @@ import {HasEditorID, ContentExtractionResult} from "./AdapterInterface";
 
 export class CKEditorAdapter extends AbstractRichtextEditorAdapter {
   editorId: string;
+  editorVersion = 4;
+  editorIdHashed: string;
 
   constructor(conf: HasEditorID) {
     super(conf);
     this.editorId = conf.editorId;
+    if(typeof window["CKEDITOR"] == 'undefined'){
+      this.editorVersion = 5;
+    }
+    this.editorIdHashed = "#" + this.editorId;
   }
 
   getEditor(): CKEDITOR.editor {
-    const ckeditor = CKEDITOR.instances[this.editorId as any];
-    if (!ckeditor) {
-      throw new Error(`Can't find ckeditor with id '${this.editorId}'`);
+    if(this.editorVersion === 4){
+      const ckeditor = CKEDITOR.instances[this.editorId as any];
+      if (!ckeditor) {
+        throw new Error(`Can't find ckeditor with id '${this.editorId}'`);
+      }
+      return ckeditor;
+    }
+    // CKEditor Version 5 detected
+    let editorDomElement = document.querySelector(this.editorIdHashed)!;
+    const isInlineEditor = editorDomElement.classList.contains('ck-editor__editable');
+    if(!isInlineEditor){
+      // Classic editor detected
+      editorDomElement = editorDomElement.nextElementSibling!.querySelector('.ck-editor__editable')!;
     }
-    return ckeditor;
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    return editorDomElement.ckeditorInstance;
   }
 
   getEditorDocument(): Document {
-    return this.getEditor().document.$ as any;
+    // for version 4
+    if(this.editorVersion == 4){
+      return this.getEditor().document.$ as any;
+    }
+    // for version 5
+    return document.querySelector(this.editorIdHashed)!.ownerDocument;
```

Comment:
You can directly use the editor instance to the containing document, so you don't have to execute same query selector again and could also avoid adding the bang operator (non-null assertion operator)

Try looking into Ck5 API:
GetDomRoot: https://ckeditor.com/docs/ckeditor5/latest/api/module_engine_view_view-View.html#function-getDomRoot

`instance.editing.view.getDomRoot()`

OR

GetEditableElement: https://ckeditor.com/docs/ckeditor5/latest/api/module_core_editor_editorui-EditorUI.html#function-getEditableElement

`instance.ui.getEditableElement()`

Reply:
`instance.editing.view.getDomRoot()` and `instance.ui.getEditableElement()` cannot be used to access the parent document.